### PR TITLE
actions: test code coverage

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add code coverage and set a 100% expectation (much less audacious than it sounds).

Need to merge others first as they add both code and coverage.

### CODE REVIEW

**Comment by wxtim** [comment]
PR #20 should the percentage.

File: cylc/rose/entry_points.py [hunk 1]
```diff
@@ -179,9 +179,6 @@ def record_cylc_install_options(
     """
     # Create a config based on command line options:
     cli_config = get_cli_opts_node(opts)
-    # Leave now if there is nothing to do:
-    if not cli_config:
-        return False
```
Comment (oliver-sanders) [inline cylc/rose/entry_points.py:184]: Remove a code branch that was impossible to activate.

File: cylc/rose/utilities.py [hunk 2]
```diff
@@ -343,13 +343,11 @@ def add_cylc_install_to_rose_conf_node_opts(rose_conf, cli_conf):
...
-    else:
-        rose_conf_opts_str = ' '.join(
-            f'{rose_conf["opts"].value} {cli_opts} (cylc-install)'.split()
-        )
-    rose_conf['opts'].value = rose_conf_opts_str
+    opts = []
+    if rose_conf['opts'].state not in ['!', '!!']:
+        opts += rose_conf["opts"].value.split()
+    opts += cli_opts.split() + ['(cylc-install)']
+    rose_conf['opts'].value = ' '.join(opts)
```
Comment (oliver-sanders) [inline cylc/rose/utilities.py:350]: Fix a minor string + list bug.

File: tests/test_config_node.py [hunk 3]
```diff
@@ -66,8 +66,11 @@ def test_get_rose_vars_from_config_node__unbound_env_var(caplog):
 @pytest.mark.parametrize(
     'rose_conf, cli_conf, expect',
     [
+        ({}, {}, '(cylc-install)'),
         # It is not given rose_node with 'opts': adds (cylc-install) to opts:
         ({}, {'opts': ''}, '(cylc-install)'),
+        # opts ignored in the config
+        ({'!opts': ''}, {}, '(cylc-install)'),
```
Comment (oliver-sanders) [inline tests/test_config_node.py:73]: Bump the coverage up a little.

**Comment by oliver-sanders** [comment]
We now have 100% coverage by line, however, the coverage figure stands at 97.44% due to "partial" coverage of some branches (i.e. some `if` statements only ever evaluated one way or another).

**Review by datamel — APPROVED** [review]
I'm happy with these changes. Thanks Oliver.

**Review by wxtim — APPROVED** [review]
Looks eminently sensible. I imagine that it's sufficiently stale that that it'll fail when merged, but happy to merge at this pre-release stage anyway.

**Pull request merged**